test/post_processors/test_collector.py

```diff
@@ -0,0 +1,282 @@
+    if collections[0]['ways'][0]['id'] == expected_asc[0]['id']:
+        assert collections[0]['ways'] == expected_asc, "If joined in ascending order, collection must match."
+    # joining happened in descending order
+    else:
+        assert collections[0]['ways'] == reverse(expected_asc), "If joined in descending order, collection must match."
+
+def test_collector_ref_matching():
+    collector = WayCollector(parser_class=Vermont125Parser)
+    collections = []
+
+    collector.parse('doesnt_exist.pbf', callback=lambda collection: collections.append(collection))
+    assert len(collections) == 3 # VT 30, VT 125, US 7
+    # Get the collection for VT 125.
+    found = False
+    for collection in collections:
+        if collection['join_type'] == 'ref' and collection['join_data'] == 'VT 125':
+            found = True
+            break
+    assert len(collection['ways']) == 5
+    assert collection['join_type'] == 'ref'
+    assert collection['join_data'] == 'VT 125'
+
+    expected_asc= [{'coords': [(-73.00009, 43.70009),
+                          (-73.0001, 43.7001),
+                          (-73.00011, 43.70011),
+                          (-73.00012, 43.70012)],
+               'id': 20004,
+               'refs': [209, 210, 211, 212],
+               'tags': {'highway': 'secondary', 'name': 'Hancock Road', 'ref': 'VT 125'}},
+              {'coords': [(-73.00012, 43.70012),
+                          (-73.00013, 43.70013),
+                          (-73.00014, 43.70014),
+                          (-73.00015, 43.70015)],
+               'id': 20003,
+               'refs': [212, 213, 214, 215],
+               'tags': {'highway': 'secondary', 'name': 'Ripton Road', 'ref': 'VT 125'}},
+              {'coords': [(-73.00015, 43.70015),
+                          (-73.00016, 43.70016),
+                          (-73.00017, 43.70017),
+                          (-73.00018, 43.70018)],
+               'id': 20000,
+               'refs': [215, 216, 217, 218],
+               'tags': {'highway': 'secondary', 'name': 'South Main Street', 'ref': 'VT 125'}},
+              {'coords': [(-73.00018, 43.70018),
+                          (-73.00019, 43.70019),
+                          (-73.0002, 43.7002),
+                          (-73.00021, 43.70021)],
+               'id': 20001,
+               'refs': [218, 219, 220, 221],
+               'tags': {'highway': 'primary', 'name': 'Court Street', 'ref': 'US 7;VT 125'}},
+              {'coords': [(-73.00021, 43.70021),
+                          (-73.00022, 43.70022),
+                          (-73.00023, 43.70023),
+                          (-73.00024, 43.70024)],
+                'id': 20002,
+                'refs': [221, 222, 223, 224],
+                'tags': {'highway': 'secondary', 'name': 'Main Street', 'ref': 'VT 125;VT 30'}}
+        ]
+    # Joining happened in ascending order.
+    if collection['ways'][0]['id'] == expected_asc[0]['id']:
```

Comment:
Same here
This should no longer be necessary? it should always be the same order and ouput